addons/website_sale_delivery/controllers/main.py

```diff
@@ -195,7 +197,44 @@ def express_checkout_process_shipping_address(self, partial_shipping_address):
                 order_sudo.currency_id,
             ),
         } for carrier in order_sudo._get_delivery_methods()],
-        key=lambda carrier: carrier['minorAmount'])
+            key=lambda carrier: carrier['minorAmount'])
+
+    @http.route('/shop/access_point/set', type='json', auth='public', methods=['POST'], csrf=False, website=True, sitemap=False)
+    def set_access_point(self, access_point_encoded):
+        order = request.website.sale_get_order()
+        if hasattr(order.carrier_id, order.carrier_id.delivery_type + '_use_locations'):
+            use_location = getattr(order.carrier_id, order.carrier_id.delivery_type + '_use_locations')
+            access_point = use_location and (json.loads(access_point_encoded) if access_point_encoded else None) or None
+            order.write({'access_point_address': access_point})
+
+    @http.route('/shop/access_point/get', type='json', auth='public', csrf=False, website=True, sitemap=False)
+    def get_access_point(self):
+        order = request.website.sale_get_order()
+        if not order.carrier_id.delivery_type:
+            return {}
+        order_location = order.access_point_address
+        if not order_location:
+            return {}
+        address = order_location['address']
+        return {order.carrier_id.delivery_type + '_access_point': address}
+
+    @http.route('/shop/access_point/close_locations', type='json', auth='public', csrf=False, website=True, sitemap=False)
```

Comment:
I wonder how safe it is to have `csrf=False`.
I assume that this function uses a secret key underneath for whatever service will be used (whether it be sendcloud or a specific carrier), meaning that this route can technically be used by anyone to retrieve information about pick up points?
Note that I have not checked the data that is returned so maybe that data is not critical at all, in that case you can disregard this comment. (I noticed it was already the case for fedex module)

Reply:
It indeed retrieves a list of pick-up points based on an address and a distance.
This data seems indeed not critical as these are not private informations.